batch/batch/driver/instance_collection/pool.py

```diff
@@ -207,12 +212,14 @@ def adjust_for_add_instance(self, instance):
         if instance.state == 'active' and instance.failed_request_count <= 1:
             self.healthy_instances_by_free_cores.add(instance)
 
-    def get_instance(self, user, cores_mcpu):
+    def get_instance(self, user: str, cores_mcpu: int, regions: List[str]):
         i = self.healthy_instances_by_free_cores.bisect_key_left(cores_mcpu)
         while i < len(self.healthy_instances_by_free_cores):
             instance = self.healthy_instances_by_free_cores[i]
             assert cores_mcpu <= instance.free_cores_mcpu
-            if user != 'ci' or (user == 'ci' and instance.region == self._ci_region):
+            if user == 'ci' and instance.region == self._ci_region:
+                return instance
+            if user != 'ci' and instance.region in regions:
```

Comment:
We don't need the CI special casing any more, right? CI can (and indeed it now does thanks to your changes here!) use the general-purpose region specification system.

Reply:
Great, I look forward to approving the removal!